File: 46-permutations/46-permutations.cpp
class Solution {
public:
    vector<vector<int>> permute(vector<int> nums) {
        int n = nums.size();
        sort(nums.begin(),nums.end());
        vector<vector<int>> ans;
        do{
            ans.push_back(nums);
        }while(next_permutation(nums.begin(),nums.end()));
        
        return ans;
    }
};

File: 46-permutations/46-permutations.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        ans = []
        N = len(nums)
        def solve(idx):
            if idx==N:
                ans.append(list(nums))
            else:
                for i in range(idx,N):
                    nums[i],nums[idx]=nums[idx],nums[i]
                    solve(idx+1)
                    nums[i],nums[idx]=nums[idx],nums[i]
                    
        
        solve(0)
        return ans